Druid-Tests/DruidTestRTLCompiler.class.st

```diff
@@ -3165,40 +3141,40 @@ DruidTestRTLCompiler >> gen_primitiveNewWithArg [
 	self CmpCq: 0 R: Extra3Reg.
 	jump5 := self JumpZero: 0.
 	self MoveMw: 32 r: SPReg R: Extra3Reg.
+	self MoveR: Extra3Reg R: SendNumArgsReg.
+	self AddCq: 8 R: SendNumArgsReg.
+	self MoveMw: 32 r: SPReg R: Extra3Reg.
 	self MoveR: Extra3Reg R: Extra2Reg.
 	self AddCq: 8 R: Extra2Reg.
-	self MoveMw: 32 r: SPReg R: Extra3Reg.
-	self MoveR: Extra3Reg R: TempReg.
-	self AddCq: 8 R: TempReg.
 	self MoveCq: 8 R: Extra1Reg.
 	self MoveMw: 8 r: SPReg R: Extra3Reg.
 	self MulR: Extra3Reg R: Extra1Reg.
```

Comment:
Here we are Multiplying by a known constant (8).
Of course we cannot generate a multiply by a constant because we don't have a MulCq:R: instruction.
But we have a bit shift (and 8 is power of 2 ;))